src/sbt-test/sbt-auto-build/test-source-generation-older-sbt/build.sbt

```diff
@@ -15,7 +15,10 @@ lazy val root = (project in file("."))
         !updated.contains(s"Copyright ${java.time.Year.now()} HM Revenue & Customs")
       }
       val failed = sources.collect { case source if noHeader(source) => source }
-      if (failed.nonEmpty)
-        sys.error(s"Header was not added to ${failed.mkString(",")}")
+      val oldDateUpdated = Source.fromFile("app/OldController.scala").getLines().mkString.contains(s"Copyright ${java.time.Year.now()} HM Revenue & Customs")
+      if (failed.nonEmpty || oldDateUpdated) {
```

Comment:
I think handling the two error case separately will be clearer:
```scala
if (failed.nonEmpty)
  sys.error(s"Header was not added to ${failed.mkString(",")}")
if (oldDateUpdated)
 sys.error(""Header date was updated in app/OldController.scala")
```

Reply:
I thought of that but then if both errors occur then only failed.nonEmpty will be reported as sys.error throws an exception.